legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/ServiceLSPGrammarExtension.java

```diff
@@ -52,6 +55,168 @@ public class ServiceLSPGrammarExtension extends AbstractSectionParserLSPGrammarE
     private static final String RUN_LEGACY_TESTS_COMMAND_ID = "legend.service.runLegacyTests";
     private static final String RUN_LEGACY_TESTS_COMMAND_TITLE = "Run legacy tests";
 
+    private static final ImmutableList<String> FUNCTIONS_TRIGGERS = Lists.immutable.with("->");
```

Comment:
Two comment on these:
- Function suggestion are not only for Services
- Im not sure the list below is helpful... We will end suggesting incompatible functions, and that can be overwhelming more that helpful.

Reply:
- Is the list of functions in Pure syntax the exact same one? If yes I agree we can make it available to both.
- Yes, we discussed it with Pierre, the refinement will require partial compilation to determine the type of the argument and propose a subset of appropriate functions. As a user I would think it's helpful to have the list of functions somewhere (even if it's a superset of what the context allows), because none really knows what functions are available in the language.

Reply:
They are the same grammar, and Service is on a deprecated path in preference of function.

I discussed with Pierre yest, and he mention he is adding few things to make this possible.  

For now, lets add it as it, and we can refactor, and move and improve the code as needed later.